Add autotest for key verification and fix several edge-cases

TobiasFella opened the pull request.

TobiasFella committed:
Add autotest for key verification and fix several edge-cases

CarlSchwan reviewed (approved):
:heart:

KitsuneRal commented on lib/keyverificationsession.h:
```diff
@@ -19,6 +19,11 @@ struct QUOTIENT_API EmojiEntry {
+public:
+    bool operator==(const EmojiEntry& rhs) const {
+        return emoji == rhs.emoji && description == rhs.description;
+    }
```
```suggestion
    bool operator==(const EmojiEntry& rhs) const = default;
```

KitsuneRal commented on autotests/testkeyverification.h:
```diff
@@ -0,0 +1,18 @@
+// SPDX-FileCopyrightText: 2022 Tobias Fella <fella@posteo.de>
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+#pragma once
+
+#include <QTest>
+#include "testutils.h"
+
+class TestKeyVerificationSession : public QObject
+{
+    Q_OBJECT
+
+private Q_SLOTS:
+    void testVerification();
+
+};
+
```
You don't need to add a separate `.h` file here; instead, just define the class in `testkeyverification.cpp` and `#include "testverification.moc"` at the end of it (moc will auto-generate it) - see how it's done in `quotest.cpp`.

KitsuneRal reviewed (changes requested):
I trust you with the logic but suggest a couple of simplifications.

TobiasFella committed:
Update lib/keyverificationsession.h

Co-authored-by: Alexey Rusakov <Kitsune-Ral@users.sf.net>

TobiasFella committed:
Remove header for keyverificationtest

KitsuneRal committed:
KeyVerificationEvent; KeyVerificationSession::handleEvent()

Key verification events gain their own base type and
KeyVerificationSession gets a single point of entry for all kinds of
incoming events. This allows to drop a pile of `incoming*` signals in
Connection and a stack of options inside switchOnType in
processIfVerification(). KVS::handleEvent() also makes (some) allowed
state transitions a bit clearer.

KitsuneRal committed:
TestKeyVerification: Add a missing allowed state
